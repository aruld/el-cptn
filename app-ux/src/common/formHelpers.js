export const renderErrors = (error, field) => {
    return error?.fieldErrors?.[field]?.map(error => <label key={`${error}`} className="label">
        <span className="label-text-alt text-error">{error}</span>
    </label>)
}