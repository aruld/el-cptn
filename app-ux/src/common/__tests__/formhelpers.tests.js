import { renderErrors } from "../formHelpers";

describe("renderErrors", () => {

    it("should return valid error message", () => {
        const error = {
            fieldErrors: { testField: ["testError"] },
        };

        const field = "testField";
        const result = renderErrors(error, field);

        expect(result).toMatchSnapshot();
    });

    it("should return null if no error", () => {
        const error = null;
        const field = "testField";
        const result = renderErrors(error, field);
        expect(result).toBeNull();
    });

    it("should return null if no fieldErrors", () => {
        const error = {};
        const field = "testField";
        const result = renderErrors(error, field);
        expect(result).toBeNull();
    });

    it("should return null if no fieldErrors for field", () => {
        const error = {
            fieldErrors: {},
        };
        const field = "testField";
        const result = renderErrors(error, field);
        expect(result).toBeNull();
    });

    it("should return null if no fieldErrors for field", () => {
        const error = {
            fieldErrors: {
                testField: [],
            },
        };
        const field = "testField";
        const result = renderErrors(error, field);
        expect(result).toBeNull();
    });

    it("should return null if no fieldErrors for field", () => {
        const error = {
            fieldErrors: {
                testField: ["testError"],
            },
        };
        const field = "testField";
        const result = renderErrors(error, field);
        expect(result).not.toBeNull();
    });
});