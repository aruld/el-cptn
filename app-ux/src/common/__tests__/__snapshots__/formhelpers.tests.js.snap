// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderErrors should return valid error message 1`] = `
Array [
  <label
    className="label"
  >
    <span
      className="label-text-alt text-error"
    >
      testError
    </span>
  </label>,
]
`;
